### PR TITLE
Document SSO

### PR DESCRIPTION
**Pull request by GuGuss**
- Adds SSO page
- Adds shortcode wrapper `tiered-feature` for subsection banners
- Reorganizes `tiered-feature` partial into subdir

### CODE REVIEW

File: docs/src/administration/sso/_index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+{{< description >}}
+
+Once SSO is enabled with your domain `@example.com`, every user with an `@example.com` email will have to authenticate through your SSO method. That user will not be able to login or register with a password or via a federated login.
+
+## Mitigation controls
+
+If you deactivate a SSO user on your identity provider, the user will not be able to login or register on Platform.sh.
+
+If the user is already logged in, he will be deactivated after his access token has expired (generally after 1 hour), and will be logged out of Platform.sh.
```
Comment (pjcdawkins) [inline docs/src/administration/sso/_index.md:20]:
```suggestion
If the user is already logged in to Platform.sh, they will be automatically deactivated after their access token has expired (generally after 1 hour).
```

File: docs/src/administration/sso/_index.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+A deactivated user will not be able to use SSH nor Git anymore.
```
Comment (pjcdawkins) [inline docs/src/administration/sso/_index.md:22]:
```suggestion
A deactivated user will no longer be able to use SSH, Git, or other Platform.sh APIs.
```

File: docs/src/administration/sso/_index.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+title: "SSO"
```
Comment (mrkschan) [inline docs/src/administration/sso/_index.md:2]:
```suggestion
title: "Single sign-on (SSO)"
```

File: docs/src/administration/sso/_index.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+If the user is already logged in, he will be deactivated after his access token has expired (generally after 1 hour), and will be logged out of Platform.sh.
+
+A deactivated user will not be able to use SSH nor Git anymore.
+
+## Service users
+
+If you have a service user with an email address under your SSO domain `machine-user@example.com`, you can whitelist that user so that they will not be required to authenticate through your SSO method. 
+
+Whitelisted users are only allowed to login or register with a password or via a federated login (Google, GitHub, Bitbucket).
```
[the remark this answers is not in the document]
Comment (GuGuss) [reply docs/src/administration/sso/_index.md:28]: They still can use our generic Google login button if they have set up Google SSO.

File: docs/src/administration/sso/_index.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+title: "Single sign-on (SSO)"
+weight: 4
+description: |
+  Platform.sh allows you to set up mandatory SSO with a third-party identity provider (IdP) for all your users.
+sidebarTitle: "Single sign-on (SSO)"
+tier:
+  - Elite
+  - Enterprise
+---
```
Comment (chadwcarlson) [inline docs/src/administration/sso/_index.md:10]: The line 
```
layout: single
```
should be added to the front matter here.

File: docs/src/administration/sso/_index.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+If the user is already logged in to Platform.sh, they will be automatically deactivated after their access token has expired (generally after 1 hour).
+
+A deactivated user will no longer be able to use SSH, Git, or other Platform.sh APIs.
+
+## Service users
+
+If you have a service user with an email address under your SSO domain, e.g. `machine-user@example.com`, you can whitelist that user so that they will not be required to authenticate through your identity provider. 
+
+
```
Comment (Crell) [inline docs/src/administration/sso/_index.md:28]:
```suggestion
```

File: docs/src/administration/sso/_index.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+---
+title: "Single sign-on (SSO)"
+weight: 4
+description: |
+  Platform.sh allows you to set up mandatory SSO with a third-party identity provider (IdP) for all your users.
+sidebarTitle: "Single sign-on (SSO)"
```
Comment (chadwcarlson) [inline docs/src/administration/sso/_index.md:6]: You don't need to include `sidebarTitle` on pages where the sidebar is the same as `title`

File: docs/src/administration/sso/google.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Google"
+weight: 2
+description: |
+  Enforce your users to authenticate with Google.
+sidebarTitle: "Google"
```
Comment (chadwcarlson) [inline docs/src/administration/sso/google.md:6]: same here

File: docs/src/administration/sso/openid-connect.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+---
+title: "OpenID Connect"
+weight: 1
+description: |
+  Enforce your users to authenticate with your OpenID Connect provider.
+sidebarTitle: "OpenID Connect"
```
Comment (chadwcarlson) [inline docs/src/administration/sso/openid-connect.md:6]: and here

File: docs/src/administration/sso/_index.md (hunk 7 above)
Comment (chadwcarlson) [inline docs/src/administration/sso/_index.md:6]:
```suggestion

```

File: docs/src/administration/sso/google.md (hunk 8 above)
Comment (chadwcarlson) [inline docs/src/administration/sso/google.md:6]:
```suggestion
```

File: docs/src/administration/sso/openid-connect.md (hunk 9 above)
Comment (chadwcarlson) [inline docs/src/administration/sso/openid-connect.md:6]:
```suggestion
```

**Pull request merged**